protostar/cairo_testing/execution_environments/cairo_test_execution_environment.py

```diff
@@ -32,8 +38,18 @@ def __init__(
 
     async def execute(self, function_name: str) -> Any:
         with self.state.output_recorder.redirect("test"):
-            await self.execute_test_case(function_name)
-            return TestExecutionResult(execution_resources=None)
+            result = TestExecutionResult(
+                execution_resources=await self.execute_test_case(function_name)
+            )
+            try:
+                cheatable_state = self.state.starknet.state.state
+                assert isinstance(cheatable_state, CheatableCachedState)
+                ExpectsCairoCheater.assert_no_expected_calls(
+                    cheatable_state.expected_contract_calls
+                )
+            except (ProtostarException, ExpectsCheaterException) as e:
+                raise ReportedException from e
+            return result
```

Comment:
I think this part could be done with a test finish hook, like in #1412 (`ExpectEventsController`)

Reply:
Yes, I saw that too 👍